utils/tim_delay.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * tim_delay.c
```

Comment:
how about, simply, `delay.c`?

Reply:
I have named it `timer_delay.c` if that is okay. If not I will change it to `delay.c` so let me know :)

Reply:
Perfectly fine.